### PR TITLE
[SOAR-10434] - Darktrace SSL verification field

### PR DESCRIPTION
**Pull request by igorski-r7**
### Description

Describe the proposed changes:

  - Added verify SSL certificate boolean field to the plugin connection

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [X] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [X] Screenshot of job output with the plugin changes
- [X] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [X] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [X] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [X] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [X] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [X] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [X] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [X] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [X] Work fully completed
- [X] Functional
  - [X] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [X] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [X] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [X] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [X] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/darktrace/icon_darktrace/util/api.py [hunk 1]
```diff
@@ -50,8 +56,49 @@ def _get_signature(self, request: str, date: str) -> str:
...
         date = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+        headers = {
+            "DTAPI-Token": self.public_token,
+            "DTAPI-Date": date,
+            "DTAPI-Signature": self._get_signature(endpoint, date),
+        }
+        return headers
+
+    def _process_request_response(self, response: requests.Response) -> Dict[str, Any]:
+        """Returns the headers needed for all the Darktrace API requests
```
Comment (ablakley-r7) [inline plugins/darktrace/icon_darktrace/util/api.py:78]: This line is a copy of the _get_headers doc string.

**Review by cmcnally-r7 — COMMENTED** [review]
Will previous versions of this plugin still work even if this is updated with the new boolean input? 
Has the service been updated so that now it **requires** SSL verify?

File: plugins/darktrace/icon_darktrace/util/api.py [hunk 2]
```diff
@@ -50,8 +56,49 @@ def _get_signature(self, request: str, date: str) -> str:
...
+            raise PluginException(preset=PluginException.Preset.USERNAME_PASSWORD, data=response.text)
+        if response.status_code == 403:
+            raise PluginException(preset=PluginException.Preset.API_KEY, data=response.text)
+        if response.status_code == 404:
+            raise PluginException(preset=PluginException.Preset.NOT_FOUND, data=response.text)
+        if 400 < response.status_code < 500:
+            raise PluginException(preset=PluginException.Preset.UNKNOWN, data=response.text)
+        if 200 <= response.status_code < 300:
+            return response.json() if response.text else {}
+        raise PluginException(preset=PluginException.Preset.SERVER_ERROR, data=response.text)
```
Comment (ablakley-r7) [inline plugins/darktrace/icon_darktrace/util/api.py:99]: May be unlikely but should we account for 300 redirect status codes separately from server error?

File: plugins/darktrace/plugin.spec.yaml [hunk 3]
```diff
@@ -602,23 +603,30 @@ types:
...
   url:
     title: URL
-    type: string
     description: API URL
+    type: string
     required: true
+    example: https://example.com
+  ssl_verification:
+    title: Verify SSL Certificate
+    description: Indicates whether verify SSL certificate or not
```
Comment (ablakley-r7) [inline plugins/darktrace/plugin.spec.yaml:615]:
```suggestion
    description: Indicates whether to verify SSL certificate or not
```

File: plugins/darktrace/plugin.spec.yaml [hunk 4]
```diff
@@ -602,23 +603,30 @@ types:
...
+    type: boolean
+    required: true
+    default: True
```
Comment (ablakley-r7) [inline plugins/darktrace/plugin.spec.yaml:618]:
```suggestion
    default: true
```

File: plugins/darktrace/plugin.spec.yaml [hunk 5]
```diff
@@ -602,23 +603,30 @@ types:
...
+    example: True
```
Comment (ablakley-r7) [inline plugins/darktrace/plugin.spec.yaml:619]:
```suggestion
    example: true
```

**Review by cmcnally-r7 — APPROVED** [review]

**Review by ablakley-r7 — CHANGES_REQUESTED** [review]
Looks good, only a couple of suggestions.

File: plugins/darktrace/icon_darktrace/util/api.py (hunk 1 above)
Comment (igorski-r7) [reply plugins/darktrace/icon_darktrace/util/api.py:78]: Oh, yea. You're right! My bad 😄

File: plugins/darktrace/icon_darktrace/util/api.py (hunk 2 above)
Comment (igorski-r7) [reply plugins/darktrace/icon_darktrace/util/api.py:99]: Oh, yea. I will connect that with `UNKNOWN` preset then. Will that be ok? 😄

**Review by jkaluza-rapid7 — APPROVED** [review]

File: plugins/darktrace/icon_darktrace/util/api.py (hunk 2 above)
Comment (ablakley-r7) [reply plugins/darktrace/icon_darktrace/util/api.py:99]: Yeah I think that sounds good.`UKNOWN` might be better than `SERVER_ERROR`.

**Review by ablakley-r7 — APPROVED** [review]

**Pull request merged**